### ISSUE TITLE
Dependent plugin not included in generated EAR - Data Conversion Plugin

### ISSUE THREAD

**Issue by kwoodie** [TIBCOSoftware/bw6-plugin-maven#259]
Maven Plugin version: 2.1.0

Maven version: 3.3.9

Product : Either of TIBCO ActiveMatrix Businessworks 6 
Product version:  6.4.2 - Hotfix 7

Component: Maven build plugin

**Steps to reproduce the issue:**
1. Create a project that is maven enabled
2. Add an activity that leverages the TIBCO Data Conversion Plugin
3. Run the "package" goal to build the ear file or run an "install" to attempt to install the EAR file into a domain.
4. See error for deployment.

If i build the ear file manually using BusinessStudio, then I get a valid ear file that is about 3x the size and works perfectly.  If I attempt the install/package with Maven it appears the ear file is much too small and throws the below error.   

Does the Maven plugin require something specific for the data conversion plugin libraries/schemas to be included?

**Describe the results you received:**
[INFO] Deploying the Application with name -> Party with Profile -> dev.substvar
[ERROR] 
com.tibco.bw.maven.plugin.admin.client.ClientException: TIBCO-BW-RESTAPI-500400: Deployed application from the archive [Party_1.0.0.ear], however not all application instances started.  <CausedBy> WARNING: exception with message ***Error starting application [Party:1.0]*** has not been localized: TIBCO-BW-ADMIN-500444: Failed to start Application in AppNode [XXXXXX]. Check the AppNode log files for messages starting with TIBCO-THOR-FRWK, TIBCO-BW-FRWK, or TIBCO-BW-SR-FRWK for details. Application State [StartFailed], reason : TIBCO-BW-CORE-500102: Failed to install BW Module [Party.module:1.0.0.201903260132], DeploymentUnit [Party:1.0].
   <CausedBy> com.tibco.bx.core.faults.BxException: BX-600025: Create module failed for module [Party.module] version [1.0.0.201903260132] application [Party 1.0].
   <CausedBy> com.tibco.bx.core.faults.BxException: BX-600028: Adding definition failed for process [party.XXXXXXX.XXXXXXXXX].
   <CausedBy> com.tibco.bw.core.runtime.schemaresolver.SchemaNotFoundException: TIBCO-BW-SCHEMA-RESOLVER-500101: Schema not found.  Schema [namespace=http://xmlns.tibco.com/dataconversion/6a0e789a-2800-4f56-8119-065135aca34e, location=null, baseURI=bundleentry://660.fwk428696898/Processes/party/mf/addParty/OrgMapMfToREST.bwp] not found in cache for the BW Module [Party.module:1.0.0.201903260132].  The SchemaCacheService operation [LocXsdCacheService.getCache] returned "null". (500)
        at com.tibco.bw.maven.plugin.admin.client.RemoteDeployer.deployApplication(RemoteDeployer.java:401)
        at com.tibco.bw.maven.plugin.admin.client.RemoteDeployer.addAndDeployApplication(RemoteDeployer.java:259)
        at com.tibco.bw.maven.plugin.application.BWEARInstallerMojo.execute(BWEARInstallerMojo.java:180)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at 

**Describe the results you expected:**
Expect the ear to build the same as running the "Export EAR" option from Business Studio.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by SujataDarekar** [comment]
@kwoodie Please share the project as well.

**Comment by rubirada** [comment]
@kwoodie We have verified the Data Conversion project with Maven 2.1.0. and successfully generated EAR using maven package goal. Attaching the sample project.
[V9.zip](https://github.com/TIBCOSoftware/bw6-plugin-maven/files/3013062/V9.zip)

**Comment by kwoodie** [comment]
The project is sensitive private information, therefore I cannot share that on GitHub.  I will take the sample project and see if I can reproduce.

**Comment by kwoodie** [comment]
If I wanted to share the plugin, is there a secure way through TIBCO Support?  I work for a current customer of TIBCO.

**Comment by HiteshNembhwani** [comment]
sure @kwoodie, that works if you have a TIBCO Support account. We'd still request to give it a try with a sample project and if that doesn't work, we can try with your as-is project via Support. Please mention on the SR (Support Request) the Maven issue number and it is only created to share the project with Maven Engg team, thanks.

**Comment by kwoodie** [comment]
Not sure what the issue was.  I cleaned things up a bit with mappings and re-generated.  things worked fine with my project after that.  Thanks for taking a look.

**Comment by kwoodie** [comment]
I found the root cause of this issue.  It appears that for some reason that Copybook generated schemas get added to the .gitignore file and are not checked into my git repo.  Locally maven builds work fine, however remote Jenkins builds fail with the missing Schema error that I created this issue about.  I corrected the .gitignore configuration and all was well on the Jenkins server when the build ran.   Thanks for writing this plugin it has made deployments a breeze.
